### PR TITLE
Lhotse support for transcribe_speech_parallel

### PR DESCRIPTION
**Pull request by nune-tadevosyan**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.
Lhotse support for transcribe_speech_parallel

**Collection**: [Note which collection this PR will affect]
ASR
# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/data/audio_to_text_dataset.py [hunk 1]
```diff
@@ -867,10 +867,16 @@ def write_on_batch_end(
                 sample = sample_id
                 if isinstance(sample, lhotse.cut.MixedCut):
                     sample = sample.first_non_padding_cut
+                if sample.recording.sources[0].source != '':
```
[the remark this answers is not in the document]
Comment (nune-tadevosyan) [reply nemo/collections/asr/data/audio_to_text_dataset.py:870]: When returning cuts metadata, `the sample.recording.sources[0].source` is always empty.

File: nemo/collections/asr/data/audio_to_text_dataset.py [hunk 2]
```diff
@@ -867,10 +867,16 @@ def write_on_batch_end(
...
+                    item["audio_filepath"] = sample.recording.sources[0].source
+                else:
+                    item["audio_filepath"] = sample.id
                 item["audio_filepath"] = sample.recording.sources[0].source
                 item["offset"] = sample.start
                 item["duration"] = sample.duration
-                item["text"] = sample.supervisions[0].text
+                item["text"] = sample.supervisions[0].text or ''
```
[the remark this answers is not in the document]
Comment (nune-tadevosyan) [reply nemo/collections/asr/data/audio_to_text_dataset.py:877]: yes

File: examples/asr/transcribe_speech_parallel.py [hunk 3]
```diff
@@ -163,6 +163,15 @@ def main(cfg: ParallelTranscriptionConfig):
     cfg.predict_ds.return_sample_id = True
     cfg.predict_ds = match_train_config(predict_ds=cfg.predict_ds, train_ds=model.cfg.train_ds)
 
+    if cfg.predict_ds.use_lhotse:
+        if cfg.predict_ds.min_duration is None or cfg.predict_ds.max_duration is None:
+            raise ValueError("min_duration and max_duration should be provided for lhotse dataloaders")
```
[the remark this answers is not in the document]
Comment (nune-tadevosyan) [reply examples/asr/transcribe_speech_parallel.py]: Done

File: nemo/collections/asr/data/audio_to_text_lhotse.py [hunk 4]
```diff
@@ -43,17 +43,22 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
...
     def __getitem__(self, cuts) -> Tuple[torch.Tensor, ...]:
         audio, audio_lens, cuts = self.load_audio(cuts)
         tokens = [
             torch.cat(
                 [
-                    torch.as_tensor(s.tokens if hasattr(s, "tokens") else self.tokenizer(s.text, s.language))
+                    torch.as_tensor(
+                        s.tokens
+                        if hasattr(s, "tokens")
+                        else s.tokens if hasattr(s, "tokens") else self.tokenizer(s.text or "", s.language)
```
[the remark this answers is not in the document]
Comment (nune-tadevosyan) [reply nemo/collections/asr/data/audio_to_text_lhotse.py]: fixed

File: nemo/collections/asr/models/configs/asr_models_config.py [hunk 5]
```diff
@@ -41,6 +41,17 @@ class ASRDatasetConfig(nemo.core.classes.dataset.DatasetConfig):
     shard_manifests: bool = False
     shuffle_n: int = 0
 
+    # lhotse support
+    use_lhotse: bool = False
+    tarred_random_access: bool = True
```
[the remark this answers is not in the document]
Comment (nune-tadevosyan) [reply nemo/collections/asr/models/configs/asr_models_config.py]: Done

File: nemo/collections/asr/models/hybrid_rnnt_ctc_bpe_models.py [hunk 6]
```diff
@@ -140,10 +140,11 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
         if config.get("use_lhotse"):
             return get_lhotse_dataloader_from_config(
                 config,
-                global_rank=self.global_rank,
-                world_size=self.world_size,
+                global_rank=self.global_rank if not config.get("do_transcribe", False) else config.get("global_rank"),
```
[the remark this answers is not in the document]
Comment (nune-tadevosyan) [reply nemo/collections/asr/models/hybrid_rnnt_ctc_bpe_models.py:146]: Done

**Review by pzelasko — APPROVED** [review]
Looking good! Thanks!
